### DISCUSSION TITLE
bug:

### DISCUSSION THREAD

**Discussion by kimusan** [folke/noice.nvim#836] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the noice.nvim docs
- [X] I have searched the existing issues of noice.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

Ubuntu 23.04 in WSL2

### Describe the bug

I wanted to move my cmdline_popup and  and the popupmenu down towards the middle of my screen. I have used the recipe for putting them together so I just changed the "rows" setting to be 20 higher. 

The cmdline_popup is moving down just fine, but the popupmenu stays in the same location. As I understand it, these values are relative to the upper row of the editor. 

Ideally I wanted to move it to the middle of the screen with the popupmenu below it (so basically 50% + 2rows). I can however not seem to find a way to do this at all.

### Steps To Reproduce

just add these settings to the Noice setup and then  :<tab><tab> to see the location of the popups.

```
views = {
        cmdline_popup = {
          position = {
            row = 25,
            col = "50%",
          },
          size = {
            width = 60,
            height = "auto",
          },
        },
        popupmenu = {
          relative = "editor",
          position = {
            row = 27,
            col = "50%",
          },
          size = {
            width = 60,
            height = 10,
          },
          border = {
            style = "rounded",
            padding = { 0, 1 },
          },
          win_options = {
            winhighlight = { Normal = "Normal", FloatBorder = "DiagnosticInfo" },
          },
        },
```

### Expected Behavior

The cmdline_popup and popupmenu are both moved 20 rows down. 

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/noice.nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by b0ae989c** [comment]
Try removing the line `relative = "editor"`. For example,
```lua
views = {
        cmdline_popup = {
          position = {
            row = 25,
            col = "50%",
          },
          size = {
            width = 60,
            height = "auto",
          },
        },
        popupmenu = {
          position = {
            row = 28,
            col = "50%",
          },
          size = {
            width = 60,
            height = 10,
          },
          border = {
            style = "rounded",
            padding = { 0, 1 },
          },
          win_options = {
            winhighlight = { Normal = "Normal", FloatBorder = "DiagnosticInfo" },
          },
        },
```

**Reply by kschulz-samsung** [reply]
This still seems to place the popup menu somewhere above the cmdline. 
![image](https://github.com/folke/noice.nvim/assets/51357000/935ffc61-e4b4-407c-adaa-60ca900ac115)

**Reply by b0ae989c** [reply]
What's your full config and terminal size? The UI may be different. My terminal is 144-by-45 and I have
![Screenshot 2024-06-04 at 11 59 36 AM](https://github.com/folke/noice.nvim/assets/141280278/ed763fcb-ddeb-467b-b955-4b2143ac7efd)
with the following config
```lua
local root = vim.fn.fnamemodify("./.repro", ":p")
for _, name in ipairs({ "config", "data", "state", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)
local plugins = {
	"folke/tokyonight.nvim",
	{
		"folke/noice.nvim",
		dependencies = "MunifTanjim/nui.nvim",
		opts = {
			views = {
				cmdline_popup = {
					position = {
						row = 25,
						col = "50%",
					},
					size = {
						width = 60,
						height = "auto",
					},
				},
				popupmenu = {
					position = {
						row = 28,
						col = "50%",
					},
					size = {
						width = 60,
						height = 10,
					},
					border = {
						style = "rounded",
						padding = { 0, 1 },
					},
					win_options = {
						winhighlight = { Normal = "Normal", FloatBorder = "DiagnosticInfo" },
					},
				},
			},
		},
	},
}
require("lazy").setup(plugins, {
	root = root .. "/plugins",
})
vim.cmd.colorscheme("tokyonight")
```

**Reply by mazulo** [reply]
I was banging my head against the desk for quite some time already trying to make the cmdline popup be centered at the middle of the screen when I finally saw your suggestion @b0ae989c. Thanks, it worked! The only thing different I also needed to do was to remove `command_palette = true` from presets for that to work.

**Comment by folke** [comment]
Please use discussions for user questions....

**Reply by kschulz-samsung** [reply]
Sorry, I was not aware of the discussion area at that time
